integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala

```diff
@@ -383,6 +384,39 @@ class CGDataMapTestCase extends QueryTest with BeforeAndAfterAll {
       sql("select * from normal_test where name='n502670' and city='c2670'"))
   }
 
+  test("test invisible datamap during query") {
```

Comment:
This test case shouldn't test the invisible/visible table directly, can you add some test case for checking invisible/visible table? Are there any performance report for this?

Reply:
What's your proposal?
The invisible datamap is a temporary status for a datamap. It indicates that the query will not use the datamap for now.
See Line404 for usage.

Reply:
There is no need to add a performance report here.
You can explain the query before/after change the visibility of the datamap, the output should be different.

Reply:
I think we should add some validate in test case, which ensure the function correct.

Reply:
Yes. We can validate it by validating the output of explain.

Reply:
@xubo245 Will you work on it or just raise a jira?